File: src/main/java/es/developer/achambi/coreframework/threading/ResponseHandlerDecorator.java
package es.developer.achambi.coreframework.threading;

public abstract class ResponseHandlerDecorator<T> extends ResponseHandler<T> {
    private ResponseHandler<T> responseHandler;

    public ResponseHandlerDecorator( ResponseHandler<T> responseHandler ) {
        this.responseHandler = responseHandler;
    }

    @Override
    public void onSuccess(Response<T> response) {
        responseHandler.onSuccess( response );
    }

    @Override
    public void onError(Error error) {
        responseHandler.onError( error );
    }
}

File: src/main/java/es/developer/achambi/coreframework/threading/ResponseHandlerDecorator.kt
package es.developer.achambi.coreframework.threading


abstract class ResponseHandlerDecorator<T>(private val responseHandler: ResponseHandler<T>)
    : ResponseHandler<T> {

    override fun onSuccess(response: T) {
        responseHandler.onSuccess(response)
    }

    override fun onError(error: Error) {
        responseHandler.onError(error)
    }
}
